From 8c6ab5e1237be8a12633bd31a760808a0425ee21 Mon Sep 17 00:00:00 2001
From: Sean Mooney <work@seanmooney.info>
Date: Wed, 28 Oct 2020 03:38:23 +0000
Subject: [PATCH] add vnic type vdpa

This change extends the portbindings api extention
definition to add a new vnic typew "vdpa" to model
a vhost-vdpa simiar to the exisign vhost-user and
virtio-forwarder vnic types.

Change-Id: I01083fb8192b2ef00934af619576889b881c1249
---
 neutron_lib/api/definitions/portbindings.py        |  8 +++++++-
 2 files changed, 21 insertions(+), 1 deletion(-)
 create mode 100644 releasenotes/notes/vhost-vdpa-cc35f8d0ff9b3c4a.yaml

diff --git a/neutron_lib/api/definitions/portbindings.py b/neutron_lib/api/definitions/portbindings.py
index 63dd83a..8b51021 100644
--- a/neutron_lib/api/definitions/portbindings.py
+++ b/neutron_lib/api/definitions/portbindings.py
@@ -117,9 +117,15 @@ VNIC_BAREMETAL = 'baremetal'
 VNIC_DIRECT_PHYSICAL = 'direct-physical'
 VNIC_VIRTIO_FORWARDER = 'virtio-forwarder'
 VNIC_SMARTNIC = 'smart-nic'
+# - vdpa:  The vHost-vdpa transport is a new vHost backend type introduced
+#          in qemu 5.1. vHost-vdpa leverages the vdpa framework introduced in
+#          kernel 5.7 to implement  a vhost offload of a standard virtio-net
+#          interface to a software or hardware backend.
+VNIC_VHOST_VDPA = 'vdpa'

 VNIC_TYPES = [VNIC_NORMAL, VNIC_DIRECT, VNIC_MACVTAP, VNIC_BAREMETAL,
-              VNIC_DIRECT_PHYSICAL, VNIC_VIRTIO_FORWARDER, VNIC_SMARTNIC]
+              VNIC_DIRECT_PHYSICAL, VNIC_VIRTIO_FORWARDER, VNIC_SMARTNIC,
+              VNIC_VHOST_VDPA]

 # VIF_DETAILS_CONNECTIVITY: Indicates what kind of connectivity the network
 #                           back-end provides: L2, L3 or not specified.
--
2.17.1
